raise AttributeError if trying to access undefined variable

fsboehme opened the pull request:
so that `getattr(settings, 'VARIABLE', default)` behaves as it should – returning the default rather than the empty object

fsboehme committed:
raise AttributeError if trying to access undefined variable so that `getattr(settings, 'VARIABLE', default)` behaves as it should

charlsagente commented:
LGTM, let me test it and I'll let you know

charlsagente commented on python_settings/__init__.py:
```diff
@@ -121,6 +121,8 @@ def __getattr__(self, item):
                     "Exception: %s - %s" % (type(ex), ex.__repr__())
                 )
+        if get_attr is empty:
+            raise AttributeError
```
cool! but instead of just raising the exception, can you provide more information to the user of what is happening?
Something like: "settings object has no attribute 'WRONG_ATTRIBUTE'"

And also can you add a unit test to add coverage to your new line?

fsboehme replied:
the latest commit should take care of this

fsboehme committed:
fix UserSettingsHolder.__setattr__

fsboehme committed:
raise AttributeError where it occurs; unit test

fsboehme commented:
also fixed `UserSettingsHolder.__setattr__` which I noticed was throwing an error when I ran the existing tests. they're all passing now.

btw – none of the tests seem to hit the `LazyInitializationImproperlyConfigured` in that `__getattr__` method I changed. i'm not sure when that should occur, so i'll leave that one to you.

charlsagente commented:
Awesome!, thanks for your contribution.
